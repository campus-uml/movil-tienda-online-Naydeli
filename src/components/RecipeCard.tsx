import React, { useState, useEffect } from "react";
import recipeImage1 from "../assets/recipe1.jpeg";
import recipeImage2 from "../assets/recipe2.jpeg";
import recipeImage3 from "../assets/recipe3.jpeg";

// Lista de imágenes con su título, descripción y duración
const slides = [
  { image: recipeImage1, title: "Ship at Your Home", text: "The products you order will be delivered to your address", duration: 5000 },
  { image: recipeImage2, title: "Healthy Meals", text: "Discover easy and healthy recipes for every day.", duration: 5000 },
  { image: recipeImage3, title: "Online Shopping", text: "You can shopping anytime, anywhere that you want.", duration: 5000 },
];

const RecipeCard: React.FC = () => {
  const [currentIndex, setCurrentIndex] = useState(0);

  // Función para cambiar a la siguiente diapositiva
  const nextSlide = () => {
    setCurrentIndex((prevIndex) => (prevIndex + 1) % slides.length);
  };

  // Función para cambiar a la diapositiva anterior
  const prevSlide = () => {
    setCurrentIndex((prevIndex) => (prevIndex - 1 + slides.length) % slides.length);
  };

  useEffect(() => {
    const interval = setInterval(() => {
      setCurrentIndex((prevIndex) => (prevIndex + 1) % slides.length);
    }, slides[currentIndex].duration);

    return () => clearInterval(interval);
  }, [currentIndex]);

  return (
    <div className="flex flex-col items-center justify-between  bg-gray-50 px-2 lg:px-0 min-h-[80vh] max-w-4xl mx-auto w-full">
      {/* Barra de progreso tipo "Estados de WhatsApp" */}
      <div className="flex gap-1 w-full max-w-3xl px-2">
        {slides.map((_, index) => (
          <div key={index} className="h-1 flex-1 bg-gray-300 rounded overflow-hidden">
            <div
              className={`h-full ${currentIndex === index ? "bg-green-500 animate-progress" : "bg-gray-300"}`}
              style={{ animationDuration: `${slides[currentIndex].duration}ms` }}
            ></div>
          </div>
        ))}
      </div>

      {/* Contenido dinámico */}
      <div className="text-center w-full lg:w-3/5 mt-6 px-2 relative flex flex-col items-center justify-center">
        {/* Imagen actual */}
        <img
          src={slides[currentIndex].image}
          alt="Cooking Illustration"
          className="w-72 h-72 object-cover mx-auto rounded-lg transition-all duration-500" // Imagen más pequeña y centrada
        />

        {/* Texto dinámico */}
        <h2 className="text-2xl lg:text-3xl font-bold text-gray-800 mt-4">{slides[currentIndex].title}</h2>
        <p className="text-gray-600 text-sm lg:text-lg mt-2">{slides[currentIndex].text}</p>

        {/* Agregar área clickeable a los bordes */}
        <div
          className="absolute inset-y-0 left-0 w-1/6 cursor-pointer"
          onClick={prevSlide}
        ></div>
        <div
          className="absolute inset-y-0 right-0 w-1/6 cursor-pointer"
          onClick={nextSlide}
        ></div>
      </div>

      {/* Botón visible en la parte inferior */}
      <div className="w-full lg:w-3/5 px-4 mb-4">
        <button className="w-full lg:w-1/3 bg-green-500 hover:bg-green-600 text-white font-semibold py-3 px-2 rounded-xl transition-all duration-300">
          Shopping Now
        </button>
      </div>

      {/* Estilos para la animación de progreso */}
      <style>{`
        @keyframes progress {
          from { width: 0%; }
          to { width: 100%; }
        }
        .animate-progress {
          animation: progress linear;
        }
      `}</style>
    </div>
  );
};

export default RecipeCard;
