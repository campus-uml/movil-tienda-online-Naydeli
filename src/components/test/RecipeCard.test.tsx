import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it,test, expect } from 'vitest';
import '@testing-library/jest-dom';
import RecipeCard from "../RecipeCard";

describe("RecipeCard Component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<RecipeCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the Shopping Now button", () => {
    render(<RecipeCard />);
    const buttonElement = screen.getByText(/Shopping Now/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it("renders the Cooking Illustration image", () => {
    render(<RecipeCard />);
    const imageElement = screen.getByAltText(/Cooking Illustration/i);
    expect(imageElement).toBeInTheDocument();
  });

  

  
  
});