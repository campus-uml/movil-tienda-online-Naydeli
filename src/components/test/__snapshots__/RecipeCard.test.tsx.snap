// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecipeCard Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-between bg-gray-50 px-2 lg:px-0 min-h-[80vh] max-w-4xl mx-auto w-full"
  >
    <div
      class="flex gap-1 w-full max-w-3xl px-2"
    >
      <div
        class="h-1 flex-1 bg-gray-300 rounded overflow-hidden"
      >
        <div
          class="h-full bg-green-400 animate-progress"
          style="animation-duration: 5000ms;"
        />
      </div>
    </div>
    <div
      class="text-center w-full lg:w-3/5 mt-6 px-2 relative flex flex-col items-center justify-center"
    >
      <img
        alt="Cooking Illustration"
        class="w-72 h-75 object-cover mx-auto rounded-lg transition-all duration-500"
        src="/src/assets/recipe2.jpeg"
      />
      <h2
        class="text-2xl lg:text-3xl font-bold text-gray-800 mt-4"
      >
        Healthy Meals
      </h2>
      <p
        class="text-gray-600 text-sm lg:text-lg mt-2"
      >
        Discover easy and healthy recipes for every day.
      </p>
      <div
        class="absolute inset-y-0 left-0 w-1/6 cursor-pointer"
      />
      <div
        class="absolute inset-y-0 right-0 w-1/6 cursor-pointer"
      />
    </div>
    <div
      class="w-full lg:w-3/5 px-4 mb-4"
    >
      <button
        class="w-full lg:w-1/3 bg-green-600 hover:bg-green-600 text-white font-semibold py-3 px-2 rounded-xl transition-all duration-300"
      >
        Shopping Now
      </button>
    </div>
    <style>
      
        @keyframes progress {
          from { width: 0%; }
          to { width: 100%; }
        }
        .animate-progress {
          animation: progress linear;
        }
      
    </style>
  </div>
</DocumentFragment>
`;
